client/src/cli/commands/logs.py

```diff
@@ -0,0 +1,100 @@
+import argparse
+import asyncio
+from collections.abc import AsyncGenerator
+from enum import Enum
+from pprint import pp
+from typing import Any
+from urllib.parse import urljoin
+
+import aiohttp
+
+import openapi_client
+
+from .util import _make_api_client, handle_api_exception, with_job_mgmt_api
+
+
+class LogCommands(Enum):
+    NAMESPACE = "namespace"
+    UID = "uid"
+    TAIL = "tail"
+    STREAM = "stream"
+
+    @classmethod
+    def values(cls) -> list[str]:
+        return [member.value for member in cls]
+
+
+def sanitize_log_params(args: argparse.Namespace) -> dict[str, Any]:
+    return {
+        k: v
+        for k, v in vars(args).items()
+        if v is not None and k in LogCommands.values()
+    }
+
+
+def stringify_dict(d: dict[str, Any]) -> dict[str, str]:
+    return {k: str(v) for k, v in d.items()}
+
+
+@with_job_mgmt_api
+def logs(client: openapi_client.JobManagementApi, args: argparse.Namespace) -> None:
+    params = sanitize_log_params(args)
+    resp = client.logs_jobs_uid_logs_get(**params)
+    pp(resp)
+
+
+async def async_generate_log_lines(
+    client: openapi_client.JobManagementApi, params: dict
+) -> AsyncGenerator[str, None]:
+    url = urljoin(client.api_client.configuration.host, f"jobs/{params['uid']}/logs")
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url, params=stringify_dict(params)) as response:
+            async for line in response.content.iter_any():
+                yield line.decode().strip()
```

Comment:
I've evaluated `httpx` and `aiohttp` for this. 
The latter seems to be better because of smaller memory footprint and better performance (see [here](https://oxylabs.io/blog/httpx-vs-requests-vs-aiohttp).

Additionally, `aiohttp` is already a dependency on the client side (we only use `httpx` server side).

The code of both libraries is nearly identical, though `aiohttp` loads the coercion of the `params` values to `str` to us and cannot handle Python objects which was possible with `httpx`.

Reply:
Sorry to sweep in late, but there's an easier solution to the problem that doesn't require an extra dependency: the generated API client has `_without_preload_content` methods that can return a response object, which can return the response body as a stream.

Example:


```python
    with _make_api_client() as api:
        client = openapi_client.JobManagementApi(api)
        try:
            resp = client.logs_jobs_uid_logs_get_without_preload_content(
                uid=args.uid,
                namespace=args.namespace,
                stream=True,
            )
            for line in resp.stream():
                print(line.decode("utf-8"), end="")
```

Reply:
Applied in 8f9a6a8fafe321e660690f8e88f613c59501823f.